moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1502,6 +1502,9 @@ class RobotState
   /** \brief Return the sum of joint distances to "other" state. Only considers active joints. */
   double distance(const RobotState& other, const JointModelGroup* joint_group) const;
 
+  /** \brief Return an upper bound on the rotation angle (radians) of any one link of the robot between two states. */
+  double maximumRotation(const RobotState& other) const;
+
```

Comment:
This interface requires a lot more collision checks in your approach than are actually needed, because you can't consider the rotation distances separately for the different collision bodies. The upper bound of rotation for the base links will usually be drastically lower than for the end effector links.

Do you think it might be worth the effort to compute the distances for all collision bodies individually and discretize their path based on these values instead? The obvious downside would be the additional complexity of constructing possibly even more robot states and selectively checking only a subset of bodies in each one.
I'm asking because I work with robots with 50+ links and bimanual motion, so checking one arm for collisions over and over again just because the other one moves a lot sounds like a lot of overhead. That being said, I personally do not use bullet checking at all at the moment and my setup is surely not the average use case.